lib/pbench/server/cache_manager.py

```diff
@@ -463,14 +464,14 @@ def filestream(self, path: str) -> Optional[dict]:
         info = self.get_info(file_path)
         info["stream"] = None
 
-        try:
-            if info["type"] == CacheType.FILE or path == Path("."):
+        if info["type"] == CacheType.FILE or file_path.name == self.name:
```

Comment:
Either I'm really missing something, or you've taken a wrong turn here. What is `file_path.name == self.name` supposed to accomplish? There isn't actually a `fio_config_date.tar.xz` *file* inside the `fio_config_date.tar.xz` tarball, and attempting to extract it won't accomplish anything but a `TarballUnpackError`.

The special case we want, either here or in the `inventory` API, is that with an *empty* path, we return a stream to the *file* `self.tarball_path` itself, rather than attempting to extract any tar artifacts *from* that file.

If the special case is here, we really need to *open* the file and return the stream, which means it needs to be closed somewhere. Well, then again, I suppose that after the `send_file` we ought to close the stream anyway, so maybe here it's just

```python
    if not file_path:
        info["stream"] = self.tarball_path.open("rb)
    else:
        ...
```

And on the `inventory` side something like

```python
   resp = sendfile(info["stream"], ...)
   info["stream"].close()
   return resp
```

... so we don't accumulate open streams. (Presumably the stream is eventually closed and the buffer freed after going out of scope, but Python doesn't give us any way to manage how many we might accumulate before that happens and it'd be safer not to rely on that.)